### PR TITLE
Fix photographing gradients

### PR DESCRIPTION
**Pull request by kugamo**
## What Does This PR Do

Fixes weird photos of people with hair gradients.

Fixes: https://github.com/ParadiseSS13/Paradise/issues/17116

## Why It's Good For The Game

Now you can take pictures of people with fashionable hair gradients again and your eyes wont bleed from looking at the result.

## Images of changes

https://user-images.githubusercontent.com/16618648/179534527-e241abb5-4ed8-4482-97dc-a271817a50fb.mp4

## Changelog
:cl:
fix: Fixed photos of people with hair gradients showing up wierd.
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
image
image

I worry about this given the frequency that `update_hair()` gets called.
```dm
//This proc is called whenever someone clicks an inventory ui slot.
/mob/proc/attack_ui(slot)
	var/obj/item/W = get_active_hand()

	if(istype(W))
		equip_to_slot_if_possible(W, slot)
	else if(!restrained())
		W = get_item_by_slot(slot)
		if(W)
			W.attack_hand(src)

	if(ishuman(src) && W == src:head)
		src:update_hair()
		src:update_fhair()
```

You can easily spam an item in and out of your hair slot. Testing the impact of this now.

**Comment by hal9000PR** [comment]
please link all the issues this fixes

**Comment by Sirryan2002** [comment]
nitpick but can you make sure the PR fix ref is properly spaced so it auto closes on merge

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 1]
```diff
@@ -385,13 +385,17 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
...
-				img_gradient.alpha = O.h_grad_alpha
-				img_gradient.color = COLOR_MATRIX_OVERLAY(O.h_grad_colour)
-				img_gradient.pixel_x = O.h_grad_offset_x
-				img_gradient.pixel_y = O.h_grad_offset_y
-				img_gradient.blend_mode = BLEND_INSET_OVERLAY
-				MA.overlays += img_gradient
+				var/icon/icn_alpha_mask = icon(gradient.icon, gradient.icon_state)
+				var/icon/icn_gradient = icon(gradient.icon, "full")
+				var/list/icn_color = ReadRGB(O.h_grad_colour)
+				icn_gradient.MapColors(rgb(icn_color[1],0,0), rgb(0,icn_color[2],0), rgb(0,0,icn_color[3]))
```
Comment (farie82) [inline code/modules/mob/living/carbon/human/update_icons.dm:391]:
```suggestion
				icn_gradient.MapColors(rgb(icn_color[1], 0, 0), rgb(0, icn_color[2], 0), rgb(0, 0, icn_color[3]))
```

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 2]
```diff
@@ -385,13 +385,17 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
...
+				icn_gradient.ChangeOpacity(O.h_grad_alpha/255)
```
Comment (farie82) [inline code/modules/mob/living/carbon/human/update_icons.dm:392]:
```suggestion
				icn_gradient.ChangeOpacity(O.h_grad_alpha / 255)
```

**Comment by AffectedArc07** [comment]
`first_male_slime.txt` has a trailing newline, please remove for name files.

**Comment by AffectedArc07** [comment]
Also config changes committed to internal repo.

**Review by farie82 — APPROVED** [review]

**Pull request merged**